### PR TITLE
fix(popConfirm): visible prop compatible

### PR DESCRIPTION
**Pull request by linhf123**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog
PopConfirm visible compatible

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      visible compatible     |
| 🇨🇳 Chinese |      兼容 visible     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at dcad56d</samp>

Use `useMergedState` hook to manage `Popconfirm` open state and warn about conflicting props. This change aligns the `Popconfirm` API with the `Popover` API.

### <samp>🤖 Generated by Copilot at dcad56d</samp>

*  Add fallback values for `useMergedState` hook arguments in `Popconfirm` component to support legacy props `visible` and `defaultVisible` ([link](https://github.com/ant-design/ant-design/pull/45702/files?diff=unified&w=0#diff-36878e30875e0d3ac3e1c9a9d6bebbcb023b8197cddd259fb7bc00976d795285L58-R59))
*  Warn users in console if both legacy and new props are provided for controlling `Popconfirm` open state ([link](https://github.com/ant-design/ant-design/pull/45702/files?diff=unified&w=0#diff-36878e30875e0d3ac3e1c9a9d6bebbcb023b8197cddd259fb7bc00976d795285L58-R59))

### CODE REVIEW

File: components/popconfirm/index.tsx [hunk 1]
```diff
@@ -55,8 +55,8 @@ const Popconfirm = React.forwardRef<TooltipRef, PopconfirmProps>((props, ref) =>
 
   const { getPrefixCls } = React.useContext(ConfigContext);
   const [open, setOpen] = useMergedState(false, {
-    value: props.open,
-    defaultValue: props.defaultOpen,
+    value: props.open ?? props.visible,
+    defaultValue: props.defaultOpen ?? props.defaultVisible,
```
Comment (afc163) [inline components/popconfirm/index.tsx:59]: 加个用例吧

Comment (afc163) [reply components/popconfirm/index.tsx:59]: onVisibleChange 还有效不

Comment (MadCcc) [reply components/popconfirm/index.tsx:59]: 留个测试用例 +1

Comment (linhf123) [reply components/popconfirm/index.tsx:59]: > onVisibleChange 还有效不

有效的

**Review by MadCcc — APPROVED** [review]

File: components/popconfirm/index.tsx (hunk 1 above)
Comment (linhf123) [reply components/popconfirm/index.tsx:59]: 按照原先的 `open` 加了一个测试用例，发现一个问题，测试用例的里面的 className 和真实环境打印不太一样
image
image

这条命令应该一直都是正确的，因为不会出现 `ant-popover-hidden`
image

**Pull request merged**